src/System.Net.HttpListener/src/System/Net/HttpListener.cs

```diff
@@ -111,20 +99,22 @@ public ExtendedProtectionPolicy ExtendedProtectionPolicy
             }
         }
 
-        public ServiceNameCollection DefaultServiceNames
+        public ServiceNameCollection DefaultServiceNames => _defaultServiceNames.ServiceNames;
+
+        public HttpListenerPrefixCollection Prefixes
         {
             get
             {
-                return _defaultServiceNames.ServiceNames;
+                if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+                CheckDisposed();
+                if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+                return _prefixes;
```

Comment:
Could you also mark `_prefixes` as readonly?  Took me a second to realize that the reason you were able to remove the lazy initialization was because it was already being initialized in the ctor, and marking it readonly will help to a) alert future readers that it's going to be created in the ctor, and b) avoid potential problems with this change if, for example, code elsewhere was trying to null out the _prefixes such that this lazy initialization was actually important (such nulling out would then hit a compiler error with the readonly).